isis/tests/FunctionalTestsCsminit.cpp

```diff
@@ -0,0 +1,207 @@
+#include "csminit.h"
+#include "spiceinit.h"
+
+#include <QString>
+#include <iostream>
+
+#include "TestCsmPlugin.h"
+#include "Fixtures.h"
+#include "TestUtilities.h"
+#include "StringBlob.h"
+#include "FileName.h"
+
+#include <nlohmann/json.hpp>
+using json = nlohmann::json;
+
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/csminit.xml").expanded();
+
+class CSMPluginFixture : public TempTestingFiles {
+  protected:
+    const csm::Plugin *plugin;
+    Cube *testCube;
+    Pvl label;
+    QString isdPath;
+    QString filename;
+
+    void SetUp() override {
+      TempTestingFiles::SetUp();
+
+      // Create and populate default test ISD
+      json isd;
+      isd["name"] = "test_isd";
+      isd["test_param_one"] = "value_one";
+      isd["test_param_two"] = "value_two";
+
+        isdPath = tempDir.path() + "/default.json";
+        std::ofstream file(isdPath.toStdString());
+        file << isd;
+        file.flush();
+
+        std::ifstream cubeLabel("data/threeImageNetwork/cube1.pvl");
+        cubeLabel >> label;
+        testCube = new Cube();
+        filename = tempDir.path() + "/csminitCube.cub";
+        testCube->fromLabel(filename, label, "rw");
+        testCube->close();
+        
+        // There should only be one plugin, so get the first one
```

Comment:
This is not true, you should use Plugin::findPlugin to get the test plugin by name. This test shouldn't depend upon the loaded plugin environment.

Reply:
Good point, thanks!